src/main/java/io/moderne/connect/commands/GitLab.java

```diff
@@ -0,0 +1,485 @@
+                    System.err.println("The required schema is [repoName,repoBranch,desiredStyle,additionalBuildArgs,skip,skipReason]");
+                    return null;
+                }
+
+                String repoPath = values[0];
+                String branch = values[1];
+                String repoStyle = values[2];
+                String additionalBuildArgs = values[3];
+                String repoSkip = values[4];
+                String skipReason = values[5];
+
+                if (StringUtils.isBlank(repoPath)) {
+                    System.out.printf("Skipping line %d because there is an empty Git repo%n", lineNumber);
+                    lineNumber++;
+                    continue;
+                }
+                if (StringUtils.isNotBlank(prefix) && !repoPath.startsWith(prefix)) {
+                    lineNumber++;
+                    continue;
+                }
+
+                if (!StringUtils.isBlank(repoSkip) && "true".equalsIgnoreCase(repoSkip)) {
+                    System.out.printf("Skipping %s at line %d because it is marked as skipped: %s%n", repoPath, lineNumber, skipReason);
+                    lineNumber++;
+                    continue;
+                }
+                if (StringUtils.isBlank(branch)) {
+                    branch = defaultBranch;
+                }
+
+
+                buildJobs.put(String.format("build-%s", repoPath), createBuildLstJob(repoPath, branch, repoStyle, additionalBuildArgs));
+                lineNumber++;
+            }
+
+            GitLabYaml.Pipeline.PipelineBuilder builder = GitLabYaml.Pipeline.builder();
+            if (downloadCLI) {
+                builder.stage(GitLabYaml.Stage.DOWNLOAD).download(createDownloadJob());
+            }
+            return builder
+                    .stage(GitLabYaml.Stage.BUILD_LST)
+                    .jobs(buildJobs)
+                    .build();
+        }
+    }
+
+    GitLabYaml.Job createDownloadJob() {
+        String downloadURL = downloadCLIUrl;
+        if (StringUtils.isBlank(downloadURL)) {
+            downloadURL = String.format("https://pkgs.dev.azure.com/moderneinc/moderne_public/_packaging/moderne/maven/v1/io/moderne/moderne-cli-%s/%s/moderne-cli-%s-%s", platform, cliVersion, platform, cliVersion);
+        }
+        String baseCommand = String.format("curl --fail --request GET --url '%s'", downloadURL);
+        String downloadCommand;
+        if (StringUtils.isNotBlank(downloadCLITokenSecretName)) {
+            downloadCommand = String.format("%s --header 'Authorization: Bearer %s' > mod", baseCommand, variable(downloadCLITokenSecretName));
+        } else if (StringUtils.isNotBlank(downloadCLIUserNameSecretName)) {
+            downloadCommand = String.format("%s --user %s:%s > mod", baseCommand, variable(downloadCLIUserNameSecretName), variable(downloadCLIPasswordSecretName));
+        } else {
+            downloadCommand = String.format("%s > mod", baseCommand);
```

Comment:
Earlier we had issues with `mod` not containing what we expected; while the `--fail` will help, so does `--output` over `>`. 
```suggestion
            downloadCommand = String.format("%s --header 'Authorization: Bearer %s' --output mod", baseCommand, variable(downloadCLITokenSecretName));
        } else if (StringUtils.isNotBlank(downloadCLIUserNameSecretName)) {
            downloadCommand = String.format("%s --user %s:%s --output mod", baseCommand, variable(downloadCLIUserNameSecretName), variable(downloadCLIPasswordSecretName));
        } else {
            downloadCommand = String.format("%s --output mod", baseCommand);
```

If there's any redirects `--location` might help too.

Reply:
done!